sopel/tools/jobs.py

```diff
@@ -127,75 +199,284 @@ def _call(self, job):
+    :param str plugin: optional job's plugin
+    :param str label: optional job's label
+    :param handler: function to be called when the job is ready to execute
+    :type handler: :term:`function`
+    :param str doc: option job's documentation
 
     Job is a simple structure that holds information about when a function
-    should be called next. They can be put in a priority queue, in which case
-    the Job that should be executed next is returned.
+    should be called next. They are best used with a :class:`Scheduler`
+    that will manage job execution when they are ready.
 
-    Calling :meth:`next` modifies the Job object with the next time it should
-    execute. Current time is used to decide when the job should be executed
-    next so it should only be called right after the function was called.
-    """
+    The :term:`function` to execute is the ``handler``, which must be a
+    callable with this signature::
+
+        def handler(manager):
+            # perform action periodically
+            # return is optional
+
+    The ``manager`` parameter can be any kind of object; usually it's an
+    instance of :class:`sopel.bot.Sopel`.
+
+    When a job is ready, you can execute it by calling its :meth:`execute`
+    method (providing the appropriate ``manager`` argument)::
+
+        if job.is_ready_to_run(time.time()):
+            job.execute(manager)  # marked as running
+            # "next times" have been updated, the job is not running
+
+    In that case, ``execute`` takes care of the running state of the job.
+
+    Alternatively, you can use a ``with`` statement to perform action before
+    and/or after executing the job; in that case, the ``with`` statement takes
+    precedence, and the :meth:`execute` method won't interfere::
+
+        with job:
+            # the job is now running, you can perform pre-execute action
+            job.execute()  # execute the job's action, no state modification
+            # the job is still marked as "running"
+            # you can perform post-execute action
 
-    max_catchup = 5
-    """How many seconds the job can get behind.
+        # outside of the with statement, the job is not running anymore
+
+    .. seealso::
+
+        The :class:`sopel.plugins.jobs.Scheduler` class is specifically
+        designed for plugin's jobs, expecting an instance of
+        :class:`sopel.bot.Sopel` as a manager, and should be used to manipulate
+        plugin jobs.
+
+        In all other case, the :class:`sopel.tools.jobs.Scheduler` class is a
+        generic job scheduler.
 
-    This governs how much the scheduling of jobs is allowed to get behind
-    before they are simply thrown out to avoid calling the same function too
-    many times at once.
     """
+    @classmethod
+    def kwargs_from_callable(cls, handler):
+        """Generate the keyword arguments to create a new instance.
+
+        :param callable handler: callable used to generate keyword arguments
```

Comment:
I'm just curious how/if Sphinx renders the type `callable`, since below you split it into a separate `:type:` line with <tt>:term:\`function\`</tt>.

Reply:
I think it doesn't. :-/